spyder/widgets/findreplace.py

```diff
@@ -666,24 +665,31 @@ def replace_find_all(self):
         """Replace and find all matching occurrences"""
         if self.editor is None:
             return
+
         replace_text = to_text_string(self.replace_text.currentText())
         search_text = to_text_string(self.search_text.currentText())
-        re_pattern = None
         case = self.case_button.isChecked()
+        word = self.words_button.isChecked()
         re_flags = re.MULTILINE if case else re.IGNORECASE | re.MULTILINE
-        re_enabled = self.re_button.isChecked()
-        # Check regexp before proceeding
-        if re_enabled:
-            try:
-                re_pattern = re.compile(search_text, flags=re_flags)
-                # Check if replace_text can be substituted in re_pattern
-                # Fixes spyder-ide/spyder#7177.
-                re_pattern.sub(replace_text, '')
-            except re.error:
-                # Do nothing with an invalid regexp
-                return
+
+        re_pattern = None
+        if self.re_button.isChecked():
+            pattern = search_text
         else:
-            re_pattern = re.compile(re.escape(search_text), flags=re_flags)
+            pattern = re.escape(search_text)
+            replace_text = replace_text.replace('\\', r'\\')
```

Comment:
Why is that necessary?

Reply:
Maybe `replace_text = re.escape(replace_text)`?
In the same vein, is there already a test for regexp replace?

eg: 
`a123a`
search: `a(\d+)a`
replace: `b\1b`
result: `b123b`

Reply:
If a similar function doesn't exist for sub text, I would just add a comment to explain.
Maybe:
`replace_text.replace(r'\', r'\\')`
is more legible than
`replace_text.replace('\\', r'\\')`
?

Reply:
> Why is that necessary?
> Maybe replace_text = re.escape(replace_text)?

I don't know for sure if it is necessary. I was just imitating the procedures used in `replace_find_selection`. As far as I can tell, `re.escape` does exactly the same thing and I would prefer that to `replace`. Perhaps I should make the correction in `replace_find_selection` as well?

> In the same vein, is there already a test for regexp replace?

No there is not. I will add your suggested test to `test_replace_all`.

Reply:
Actually, it looks like it has to remain `replace_text.replace('\\', r'\\')`.
`re.escape` cannot be used since it escapes _all_ special characters, not just `\`.
Furthermore, `replace_text.replace(r'\', r'\\')` results in code analysis reporting "Unexpected character after line continuation character".

Reply:
Reading re.sub:

repl can be a string or a function; if it is a string, any backslash escapes in it are processed. That is, \n is converted to a single newline character, \r is converted to a carriage return, and so forth. Unknown escapes of ASCII letters are reserved for future use and treated as errors. Other unknown escapes such as \& are left alone. Backreferences, such as \6, are replaced with the substring matched by group 6 in the pattern.

It looks like you are correct. You could add a comment saying: escape backslashes as they are processed by re.sub